src/DTO/Flag.php

```diff
@@ -21,4 +21,86 @@ public function __construct(
         public int $totalShards
     ) {
     }
+
+    public static function fromJson(array $json): Flag
+    {
+        $variationType = VariationType::from($json['variationType']);
+        $variations = self::parseVariations($json['variations'], $variationType);
+        $allocations = self::parseAllocations($json['allocations']);
+
+        return new Flag(
+            $json['key'],
+            $json['enabled'],
+            $allocations,
+            $variationType,
+            $variations,
+            $json['totalShards']
+        );
+    }
+
+
+    /**
+     * @param array $variations
+     * @param VariationType $variationType
+     * @return Variation[]
+     */
+    private static function parseVariations(array $variations, VariationType $variationType): array
+    {
+        return array_map(function ($variationConfig) use ($variationType) {
+            $typedValue = $variationType === VariationType::JSON ? json_decode(
+                $variationConfig['value'],
+                true
+            ) : $variationConfig['value'];
+            return new Variation($variationConfig['key'], $typedValue);
+        },
+            $variations);
+    }
+
+    /**
+     * @param array $allocations
+     * @return Allocation[]
+     */
+    private static function parseAllocations(array $allocations): array
+    {
+        return array_map(function ($allocationConfig) {
+            $rules = array_key_exists('rules', $allocationConfig) ? array_map(function ($ruleConfig) {
+                $conditions = array_map(function ($conditionConfig) {
+                    return new Condition(
+                        $conditionConfig['attribute'],
+                        $conditionConfig['operator'],
+                        $conditionConfig['value']
+                    );
+                }, $ruleConfig['conditions']);
+                return new Rule($conditions);
+            }, $allocationConfig['rules']) : null;
```

Comment:
I'd expect `: null;` on its own line. Or maybe flip the boolean so it can come first and not be buried after the mapping

Reply:
fixed